intrinsic_and_extrinsic_optimise_real_data/src/camera_calibration_real_data.cpp

```diff
@@ -0,0 +1,422 @@
+		temp.push_back(rvec.at<double>(0, 2)); // axis-angle z
+		temp.push_back(tvec.at<double>(0, 0)); // t1
+		temp.push_back(tvec.at<double>(0, 1)); // t2
+		temp.push_back(tvec.at<double>(0, 2)); // t3
+		
+		camera_extrinsics.push_back(temp); 
+	}
+
+
+	// Display the initial camera extrinsics multidimentional array
+	for (int z = 0; z != camera_extrinsics.size(); z++)
+	{
+		std::cout << "Camera extrinsic: " << z << "\r\n";
+		std::cout << "Initial axis-angle x: " << camera_extrinsics[z][0] << "\r\n";
+		std::cout << "Initial axis-angle y: " << camera_extrinsics[z][1] << "\r\n";
+		std::cout << "Initial axis-angle z: " << camera_extrinsics[z][2] << "\r\n";
+		std::cout << "Initial t1: " << camera_extrinsics[z][3] << "\r\n";
+		std::cout << "Initial t2: " << camera_extrinsics[z][4] << "\r\n";
+		std::cout << "Initial t3: " << camera_extrinsics[z][5] << "\r\n\n";
+	}
+
+
+	// for (int z = 0; z != total_chessboards_detected; z++)
+	// {
+	// 	std::cout << "Camera extrinsic: " << z << "\r\n";
+	// 	std::cout << "Initial axis-angle x: " << camera_extrinsics[z][0] << "\r\n";
+	// 	std::cout << "Initial axis-angle y: " << camera_extrinsics[z][1] << "\r\n";
+	// 	std::cout << "Initial axis-angle z: " << camera_extrinsics[z][2] << "\r\n";
+	// 	std::cout << "Initial t1: " << camera_extrinsics[z][3] << "\r\n";
+	// 	std::cout << "Initial t2: " << camera_extrinsics[z][4] << "\r\n";
+	// 	std::cout << "Initial t3: " << camera_extrinsics[z][5] << "\r\n\n";
+	// } 
+
+
+	///////////// The optimisation code begins from here /////////////
+
+    // Set the initial values for the mutable parameters.
+    double camera_intrinsics[4] = {initial_fx, initial_fy, initial_cx, initial_cy};
+
+    // Begin building the problem
+    ceres::Problem problem;
+	std::vector<ceres::ResidualBlockId> residual_block_ids;
+
+	for (int z = 0; z != Chessboard_corners.size(); z++)
+	{
+		// std::cout << "////////////////////////// Image number: " << z << " ////////////////////////// \r\n";	
+		
+		for (int i = 0; i != Chessboard_corners[z].size(); i++)
+		{
+			double board_T_point[3] = {matrix_D[i].x, matrix_D[i].y, matrix_D[i].z}; //This is the calibration point of the format: X,Y,Z
+			double image_pixels[2] = {Chessboard_corners[z][i].x, Chessboard_corners[z][i].y}; //This is the image point of the format: u, v.
+
+			// std::cout << "Calibration point number: " << i << "\r\n";
+			// std::cout << "XYZ: " << board_T_point[0] << ", " << board_T_point[1] << ", " << board_T_point[2] << "\r\n\n";
+
+			// std::cout << "Pixel point number: " << i << "\r\n";
+			// std::cout << "pixel u: " << image_pixels[0] << "\r\n";
+			// std::cout << "pixel v: " << image_pixels[1] << "\r\n\n";
+
+			ceres::CostFunction* cost_function = ReProjectionResidual::Create(image_pixels, board_T_point);
+			ceres::ResidualBlockId block_id = problem.AddResidualBlock(cost_function, NULL, &camera_extrinsics[z][0], camera_intrinsics);
+			residual_block_ids.push_back(block_id);
```

Comment:
Pretty sure you don't need `residual_block_ids`.  Simply calling `problem.AddResidualBlock` should add the block to the problem.